Training/Medium/knight.rb

```diff
@@ -0,0 +1,33 @@
+STDOUT.sync = true # DO NOT REMOVE
+
+# Solution for https://www.codingame.com/ide/puzzle/shadows-of-the-knight-episode-1
+class Knight
+  attr_accessor :width, :height, :jumps_number, :x, :y, :search_area
+
+  def initialize
+    @width, @height = gets.split(' ').collect(&:to_i)
+    @jumps_number = gets.to_i # maximum number of turns before game over.
+    @x, @y = gets.split(' ').collect(&:to_i)
+    @search_area = [[0, 0], [@width - 1, @height - 1]]
+  end
+
+  def change_batman_position(direction)
+    search_area[1][1] = y - 1 if direction.include?('U')
+    search_area[0][1] = y + 1 if direction.include?('D')
+    search_area[1][0] = x - 1 if direction.include?('L')
+    search_area[0][0] = x + 1 if direction.include?('R')
```

Comment:
You can refactor this with a `case` statement:
```ruby
case
when direction.include?('U')
  search_area[1][1] = y - 1
when direction.include?('D')
  ...
```

Another thing that could be added is to have constants for each direction:
```ruby
UP = 'U'
DOWN = 'D'
LEFT = 'L'
RIGHT = 'R'
```
Then the case will be very descriptive and you can change the letters for directions easily:
```ruby
case
when direction.include? UP
  search_area[1][1] = y - 1
when direction.include? DOWN
  ...
```